files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -194,6 +239,14 @@ public function stream_open( $path, $mode, $options, &$opened_path ) {
 	 * @access  public
 	 */
 	public function stream_close() {
+		$this->debug( sprintf( 'stream_close => %s + %s', $this->path, $this->uri ) );
+
+		if ( $this->should_flush_empty ) {
+			$this->stream_flush();
+
+			$this->should_flush_empty = false;
```

Comment:
You already have a `$this->should_flush_empty = false;` set on one code path in `stream_flush()` and IMO this _should_ be encapsulated in `stream_flush()` and `stream_write()` and not set here. Unless there is some reason to blanket set it no matter whether `stream_flush()` succeeds?

Reply:
Ah good catch. This was a leftover from a previous attempt to trigger the flush using a destructor method which didn't really work. I'll remove this

Reply:
Removed in 98290455510c6d085003584d3ab452dfea18bc3c